src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -678,22 +678,35 @@ export class TerminalInstance implements ITerminalInstance {
 
 	public preparePathForTerminalAsync(path: string): Promise<string> {
 		return new Promise<string>(c => {
-			const hasSpace = path.indexOf(' ') !== -1;
-			if (platform.isWindows) {
-				const exe = this.shellLaunchConfig.executable;
-				if (!exe) {
+			const exe = this.shellLaunchConfig.executable;
+			if (!exe) {
+				if (platform.isWindows) {
 					c(path);
-					return;
+				} else {
+					c(this._escapeNonWindowsPath(path));
```

Comment:
Did this case come up? I think this path was here as a sanity check but shouldn't actually happen (exe should always be set here).

Reply:
tslint failed the build because [exe might be null](https://dev.azure.com/vscode/VSCode/_build/results?buildId=12488) so I moved the null check up. I could also disable the rule - your call.

Reply:
fixed